ssl.go

```diff
@@ -0,0 +1,110 @@
+//go:build !go_tarantool_ssl_disable
```

Comment:
Yes, we agreed that SSL should be enabled by default. Using the tag, we disable SSL support. It is the reason why such name is used.

A tag can be set to almost any go command:
```bash
go build -tags tag_name
go run -tags tag_name
go test -tags tag_name
```
https://pkg.go.dev/go/build#hdr-Build_Constraints

Reply:
I would discuss this point once again. One of possible reasons to use Go in various tools is portability of the resulting executable. Linking with particular version of openssl may decrease portability of the executable. So maybe it should be explicit choice of a user.

Reply:
It seems to me that it is better to provide this feature by default. This is a new functionality that we plan to support and provide. This is not an alternative to something that already exists. I don't see any reason from the library why it should be disabled.

Not every user of the library will use the feature, but the problem with linking and portability is not too difficult to notice. And then add a tag to disable it if needed.

My conclusion: from the point of view of the library, this should be enabled by default. From the user's point of view, he can easily disable it.

Reply:
'Easy to change' sometimes is not so easy. Say, software vendor A ships an application build to company B, but the operation team is actually in company C (B's integrator). Shipping a new application version means some agreement between all three on an upgrade process. It may involve a lot of people: developers, managers and operation team from different sides. Well, there may be also a pure technical complexity: the application has a component, which has a component, which actually has the problem and the software vendor needs to fix the nested component, release and deploy it, update the outer component, release and deploy it, update the application, release it and give the built to B&C. All those processes are performed by different teams (in A) with its own SLA between them.

So time between the point when one found that something is broken and actual fix may be quite large. Not taking even taking into account amount of effort of different people to do so.

(This picture is not very different from what we actually have between us and our customers.)

The risk of problem is not always compensated by simplicity of the change.

I'm not insist on staying on the most safe side here, but, please, take this picture into account. Since there is no strong arguments for enabling or disabling (only personal feeling of 'risky' and 'easy to change'), I'll leave it up to the author of the patch. Closing the discussion.